lib/pbench/agent/tool_data_sink.py

```diff
@@ -2059,26 +2042,62 @@ def main(argv):
         ToolDataSink.fetch_params(params, pbench_run)
     except Exception as ex:
         print(
-            f"Unable to fetch and decode parameter key, {param_key}: {ex}",
+            f"Unable to fetch and decode parameter key, {parsed.key}: {ex}",
             file=sys.stderr,
         )
         return 6
 
     optional_md = params.get("optional_md", dict())
 
-    func = daemon if daemonize == "yes" else driver
+    func = daemon if parsed.daemonize else driver
     ret_val = func(
         PROG,
         redis_server,
-        redis_host,
-        redis_port,
+        parsed,
         pbench_bin,
         pbench_run,
         hostname,
         tar_path,
         cp_path,
-        param_key,
         params,
         optional_md,
     )
     return ret_val
+
+
+def main(argv):
+    """Main program for the Tool Meister.
+
+    Arguments:  argv - a list of parameters
+
+                argv[1] - host name or IP address of Redis Server
+                argv[2] - port number of Redis Server
+                argv[3] - name of key in Redis Server for operational
+                          parameters
+                argv[4] - "yes" to run as a daemon
+                argv[5] - desired debug level
+
+    Returns 0 on success, > 0 when an error occurs.
+    """
+    prog = Path(argv[0])
+    PROG = prog.name
+    try:
+        parsed = Arguments(
+            host=argv[1],
+            port=int(argv[2]),
+            key=argv[3],
+            daemonize=argv[4] == "yes" if len(argv) > 4 else False,
+            level=argv[5] if len(argv) > 5 else "info",
+        )
+    except (ValueError, IndexError) as e:
+        print(f"{PROG}: Invalid arguments, {argv!r}: {e}", file=sys.stderr)
+        return 1
```

Comment:
Since we're at the outer-most level, I suspect that we should be catching all `Exception`'s -- I think that dying with a traceback should be avoided (logging it would be great; dumping it at the command line would be less so).

Reply:
I really wasn't trying to completely rearchitect the file. (Although I kinda started out in that direction, seeing Pete already doing multiple PRs overlaying this code dissuaded me from being very radical. I peeled of `main` from `start` because I was going to use `ArgumentParser` and options, but I chickened out and even backed off on that.)

Reply:
I am not touching this code in other PRs (except for the `black` work in PR #2703).